### DISCUSSION TITLE
Does IDOM support CSS modules?

### DISCUSSION THREAD

**Discussion by Archmonger** [idom-team/idom#478] [Question] [answered]
I was reading up about React and was wondering if IDOM has an integrated way of implementing CSS modules.

ex) https://www.w3schools.com/react/react_css.asp

**Comment by rmorshea** [comment, marked as answer]
You can add style to elements in two ways:

1. with a `style` tag element
2. using the `style` prop (as in React)

```python
# style element (write a CSS string)
html.style(".some-css {...}")

# define style the same as React with a camelCased dict
html.div({"style": {"backgroundColor": "red"})
```

**Reply by jgburford** [reply]
Can style be updated with hooked variables?

For example:

```python
bgstate, set_bgstate = idom.hooks.use_state("red")

html.div({"style": {"backgroundColor": bgstate})
```
I've tried this and can't get it to work. It will start red but when calling set_bgstate() it doesn't change.

Any tips?

**Reply by rmorshea** [reply]
Yup that's a bug. That's probably also why the snake game from the examples is broken right now too.

**Reply by rmorshea** [reply]
Tracking it here: https://github.com/idom-team/idom/issues/480

**Reply by rmorshea** [reply]
@jgburford without that little hint it would have taken me ages to debug the underlying issue. Thanks! :pray:

**Reply by jgburford** [reply]
@rmorshea glad I could be of some help!

**Comment by Archmonger** [comment]
I don't think it's technologically possible for IDOM to have scoped CSS, such as with [`css-modules`](https://create-react-app.dev/docs/adding-a-css-modules-stylesheet/).

The only viable solution is inline styles, as [mentioned above](https://github.com/idom-team/idom/discussions/478#discussioncomment-1240179)
